src/adapters/parsers/summary-logs/exceljs-parser.js

```diff
@@ -1,16 +1,277 @@
   }
 }
+
+const finalizeRowForCollection = (collection) => {
+  if (collection.state === 'HEADERS') {
+    return { ...collection, state: 'ROWS', currentRow: [] }
+  } else if (collection.state === 'ROWS' && collection.currentRow.length > 0) {
+    const isEmptyRow = collection.currentRow.every((val) => val === null)
+    if (isEmptyRow) {
+      return { ...collection, complete: true }
+    } else {
+      return {
+        ...collection,
+        rows: [...collection.rows, collection.currentRow],
+        currentRow: []
+      }
+    }
+  } else {
+    return collection
+  }
+}
+
+const emitCollectionsToResult = (state, collections) => {
+  for (const collection of collections) {
+    if (state.result.data[collection.sectionName]) {
+      throw new Error(`Duplicate data section name: ${collection.sectionName}`)
+    }
+    state.result.data[collection.sectionName] = {
+      location: collection.location,
+      headers: collection.headers,
+      rows: collection.rows
+    }
+  }
+}
+
+/** @type {SummaryLogParser} */
+export const parse = async (summaryLogBuffer) => {
+  const workbook = new ExcelJS.Workbook()
+  await workbook.xlsx.load(summaryLogBuffer)
+
+  let state = {
+    result: { meta: {}, data: {} },
+    activeCollections: [],
+    metadataContext: null
+  }
+
+  for (const worksheet of workbook.worksheets) {
+    const rows = []
+    worksheet.eachRow((row, rowNumber) => {
+      rows.push({ row, rowNumber })
+    })
+
+    for (const { row, rowNumber } of rows) {
+      const cells = []
+      row.eachCell({ includeEmpty: true }, (cell, colNumber) => {
+        cells.push({ cell, colNumber })
+      })
+
+      // Initialize currentRow for all active collections
+      state.activeCollections = state.activeCollections.map((c) => ({
+        ...c,
+        currentRow: []
+      }))
+
+      // Process cells with reduce
+      state = cells.reduce((acc, { cell, colNumber }) => {
+        const rawCellValue = cell.value
+        const cellValue = extractCellValue(rawCellValue)
+        const cellValueStr = cellValue?.toString() || ''
+
+        const stateAfterMetadata = processCellForMetadata(
+          cellValue,
+          cellValueStr,
+          worksheet,
+          rowNumber,
+          colNumber,
+          acc
+        )
+
+        const collectionsAfterMarkers = processDataMarker(
+          cellValueStr,
+          worksheet,
+          rowNumber,
+          colNumber,
+          stateAfterMetadata.activeCollections
+        )
+
+        const updatedCollections = collectionsAfterMarkers.map((collection) =>
+          updateCollectionWithCell(
+            collection,
+            cellValue,
+            cellValueStr,
+            colNumber
+          )
+        )
+
+        return {
+          ...stateAfterMetadata,
+          activeCollections: updatedCollections
+        }
+      }, state)
+
+      // Finalize row for each collection
+      state.activeCollections = state.activeCollections.map((c) =>
```

Comment:
Personally, I find single letter variable names esoteric and they increase cognitive load unnecessarily